src/array_api_stubs/_draft/array_object.py

```diff
@@ -343,9 +341,52 @@ def __dlpack__(
 
         Notes
         -----
+        Major DLPack versions represent ABI breaks, minor versions represent
+        ABI-compatible additions (e.g., new enum values for new data types or
+        device types).
+
+        The ``max_version`` keyword was introduced in v2023.12, and goes
+        together with the ``DLManagedTensorVersioned`` struct added in DLPack
+        1.0. This keyword may not be used by consumers for some time after
+        introduction. It is recommended to use this logic in the implementation
+        of ``__dlpack__``:
+
+        .. code:: python
+
+            if max_version is None:
+                # Keep and use the DLPack 0.X implementation
+                # Note: from March 2025 onwards (but ideally as late as
+                # possible), it's okay to raise BufferError here
+            else:
+                # We get to produce `DLManagedTensorVersioned` now
+                if max_version >= our_own_dlpack_version:
```

Comment:
nit: Can we clarify that when returning `DLManagedTensorVersioned`, the producer should fill in the version struct using `our_own_dlpack_version` not `max_version`? Put it differently, this is how we define `our_own_dlpack_version`.

Reply:
Done in commit 1db1717.